packages/pigeon/lib/dart_generator.dart

```diff
@@ -236,9 +238,13 @@ void generateDart(DartOptions opt, Root root, StringSink sink) {
       });
       indent.writeln('');
       indent.write(
-        'static ${klass.name} decode(Object message) ',
+        'static ${klass.name} decode(Object$nullTag message) ',
```

Comment:
I don't think we should change this to `Object?`.  Do you have a reason why you are doing that?